### PR TITLE
Slack Notify feature

### PR DESCRIPTION
**Pull request by w3st3ry**
Related to #1 

Signed-off-by: Valentin Pichard <w3st3ry@users.noreply.github.com>

### CODE REVIEW

**Comment by w3st3ry** [comment]
Messages are not formatted actually.
I will do a preview of a formatted message using https://api.slack.com/block-kit asap, before merging.

**Comment by nomonamo** [comment]
LGTM, don't forget to update config/README.md L51

File: pkg/notify/slack/slack.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		return
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	resp, err := sn.httpClient.Do(req)
+	if err != nil {
+		notify.WrappedSendError(Type, err.Error())
+		return
+	}
+
```
Comment (Alkorin) [inline pkg/notify/slack/slack.go]:
```suggestion
defer resp.Body.Close()
```

Comment (w3st3ry) [reply pkg/notify/slack/slack.go]: Thanks 👍

File: pkg/notify/notify.go [hunk 2]
```diff
@@ -8,20 +8,19 @@ import "github.com/ovh/utask"
...
-	// Actions represents configuration of each notify actions
+	// actions represents configuration of each notify actions
 	actions utask.NotifyActions
 )
 
 // Payload is the holder of data to be sent as a notification
 type Payload interface {
-	Message() string
-	Fields() []string
+	Message() *Message
```
Comment (nomonamo) [inline pkg/notify/notify.go]: I believe the Payload interface doesn't accomplish much anymore... why not pass around *Message directly?

File: pkg/notify/notify.go [hunk 3]
```diff
@@ -8,20 +8,19 @@ import "github.com/ovh/utask"
...
 }
 
 // NotificationSender is an object capable of sending a payload
 // over a notification channel, as determined by its implementation
 type NotificationSender interface {
-	Send(p Payload)
+	Send(p Payload, name string)
```
Comment (nomonamo) [inline pkg/notify/notify.go]: accept *Message instead of Payload

File: pkg/notify/slack/slack.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+func formatSendRequest(m *notify.Message, name string) *formattedSlackRequest {
+	var fsr formattedSlackRequest
+
+	sec := "section"
+	mrk := "mrkdwn"
+
+	// First line title
+	fsr.Blocks[0].Type = sec
```
Comment (nomonamo) [inline pkg/notify/slack/slack.go]: is Blocks != nil?

Comment (nomonamo) [reply pkg/notify/slack/slack.go]: perhaps unit-test this bit of code

Comment (w3st3ry) [reply pkg/notify/slack/slack.go]: Fixed /w dynamic and static allocation.

File: pkg/notify/notify.go (hunk 2 above)
Comment (w3st3ry) [reply pkg/notify/notify.go]: I done that because of the legacy architecture, fixed now.

File: pkg/notify/messages.go [hunk 5]
```diff
@@ -18,31 +24,31 @@ type TaskStateUpdate struct {
...
 
-// Message renders a string representation of a TaskStateUpdate
-func (tsu TaskStateUpdate) Message() string {
-	return fmt.Sprintf(
-		"#task #id:%s %s",
-		tsu.PublicID,
-		tsu.Title,
-	)
+// Message returns a task state change struct
+func (m *Message) Message() *Message {
```
Comment (nomonamo) [inline pkg/notify/messages.go]: not sure what's the use of this method anymore ?

Comment (w3st3ry) [reply pkg/notify/messages.go]: Fixed an rebased on master upstream.

**Pull request merged**